src/IECoreScene/CurvesAlgo.cpp

```diff
@@ -490,25 +369,31 @@ CurvesPrimitivePtr deleteCurves( const CurvesPrimitive *curvesPrimitive, const t
 			case PrimitiveVariable::Uniform:
 			{
 				IECore::Data *inputData = const_cast< IECore::Data * >( it->second.data.get() );
-				IECore::DataPtr ouptputData = despatchTypedData<DeleteFlaggedUniformFunctor<T> , TypeTraits::IsVectorTypedData>( inputData, deleteUniformFn );
-				outCurvesPrimitive->variables[it->first] = PrimitiveVariable( it->second.interpolation, ouptputData );
+				deleteUniformFn.setIndices( it->second.indices.get() );
+				IECoreScene::PrimitiveVariableAlgos::IndexedData outputData = dispatch( inputData, deleteUniformFn );
+				// todo update tests: setGeometricInterpretation( outputData.data.get(), getGeometricInterpretation( inputData ) );
```

Comment:
Does this comment mean we're not maintaining the interpretation currently? If so, can we take care if it in this PR? Note that you're better off transferring the interpretation within deleteUniformFn by calling `data->setInterpretation()`, because the utility methods in the comment will result in two unnecessary additional calls to `dispatch()`.

Reply:
> Does this comment mean we're not maintaining the interpretation currently?

I've corrected this in the latest tests and implementation (0c03479 &  1b9a50f )

 > Note that you're better off transferring the interpretation within deleteUniformFn by calling data->setInterpretation(), because the utility methods in the comment will result in two unnecessary additional calls to dispatch().

I attempted to move the setInterpretation into the various functors but found it more complicated in practice. I'd like to give this another as a follow up after the fixing the mesh & points bugs if that's ok?

Reply:
Sure thing.